src/coord/src/error.rs

```diff
@@ -247,6 +253,17 @@ impl fmt::Display for CoordError {
             CoordError::InvalidAlterOnDisabledIndex(name) => {
                 write!(f, "invalid ALTER on disabled index {}", name.quoted())
             }
+            CoordError::InvalidMultipleMaterialization {
+                base_source,
+                existing_index,
+                count,
+            } => {
+                write!(
+                    f,
+                    "Cannot create second materialization on {}, already materialized by {} (and {} other items)",
+                    base_source, existing_index, count - 1
+                )
+            }
```

Comment:
This is a great moment to use the `hint` and `detail` fields. I suggest something like the following:

```
ERROR: cannot re-materialize source {}
DETAIL: {PostgreSQL|SQS-enabled S3} sources can be materialized by only one set of indexes at a time. The following indexes have already materialized this source:
  index1
  index2
HINT: See the documentation at https://materialize.com/docs/wherever.
```